ihatemoney/api.py

```diff
@@ -1,62 +1,68 @@
 # -*- coding: utf-8 -*-
 from flask import Blueprint, request
-from flask_rest import RESTResource, need_auth
+from flask_restful import Resource, Api, abort
 from wtforms.fields.core import BooleanField
 
 from ihatemoney.models import db, Project, Person, Bill
 from ihatemoney.forms import (ProjectForm, EditProjectForm, MemberForm,
                               get_billform_for)
 from werkzeug.security import check_password_hash
+from functools import wraps
 
 
 api = Blueprint("api", __name__, url_prefix="/api")
+restful_api = Api(api)
 
 
-def check_project(*args, **kwargs):
+def need_auth(f):
     """Check the request for basic authentication for a given project.
 
-    Return the project if the authorization is good, False otherwise
+    Return the project if the authorization is good, abort the request with a 40x otherwise
     """
-    auth = request.authorization
-
-    # project_id should be contained in kwargs and equal to the username
-    if auth and "project_id" in kwargs and \
-            auth.username == kwargs["project_id"]:
-        project = Project.query.get(auth.username)
-        if project and check_password_hash(project.password, auth.password):
-            return project
-    return False
-
-
-class ProjectHandler(object):
-
-    def add(self):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        auth = request.authorization
+        project_id = kwargs.get("project_id")
+
+        if auth and project_id and auth.username == project_id:
+            project = Project.query.get(project_id)
```

Comment:
that's probably me being silly, but in the previous version of this code, I was using `auth.username` here rather than the direct input from the user, because I tend to trust less the input from the user.

If for whatever reason the `==` part evaluates to `True`, it incidentally opens a breach to do queries with other unintended `ids`. I cannot think of an use for this right now, but I thought I would share my thoughts on this. What's your take on this?

Reply:
Yeah I see your point but ``auth.username`` is also a direct user input, isn't it ?